presto-main/src/main/java/com/facebook/presto/server/ServerMainModule.java

```diff
@@ -304,6 +305,7 @@ protected void setup(Binder binder)
         binder.bind(TaskManagementExecutor.class).in(Scopes.SINGLETON);
         binder.bind(SqlTaskManager.class).in(Scopes.SINGLETON);
         binder.bind(TaskManager.class).to(Key.get(SqlTaskManager.class));
+        binder.bind(ContinuousBatchTaskStatusFetcher.class).in(Scopes.SINGLETON);
```

Comment:
let's inject it in SqlStageExecution constrcutor and pass it from there to HttpRemoteTaskFactor > HttpRemoteTask.